kube/src/service/builder_ext.rs

```diff
@@ -0,0 +1,37 @@
+// Borrowing from Tonic https://git.io/JtwWj
+// MIT Copyright (c) 2020 Lucio Franco
+use tower::{
+    layer::{util::Stack, Layer},
+    util::Either,
+    ServiceBuilder,
+};
+
+pub(crate) trait ServiceBuilderExt<L> {
+    fn optional_layer<T>(self, l: Option<T>) -> ServiceBuilder<Stack<OptionalLayer<T>, L>>;
+}
+
+impl<L> ServiceBuilderExt<L> for ServiceBuilder<L> {
+    fn optional_layer<T>(self, inner: Option<T>) -> ServiceBuilder<Stack<OptionalLayer<T>, L>> {
+        self.layer(OptionalLayer { inner })
+    }
+}
```

Comment:
nit: Do we technically need to define a `ServiceBuilderExt` trait for this if it's all private?

Reply:
We need it to add the method to `ServiceBuilder` so we can use it in `kube/src/service/mod.rs`.

We can't just `impl<L> ServiceBuilder<L> {}` because `ServiceBuilder` is not defined in this crate.

Reply:
ah, orphan rule again.